app/controllers/notes_controller.rb

```diff
@@ -255,15 +255,52 @@ def destroy
   ##
   # Return a list of notes matching a given string
   def search
-    # Check the arguments are sane
-    raise OSM::APIBadUserInput, "No query string was given" unless params[:q]
+    # Filter either by the name or the id of the user
+    if params[:display_name]
+      @user = User.find_by(:display_name => params[:display_name])
+    elsif params[:id]
+      @user = User.find_by(:id => params[:id])
+    end
 
-    # Get any conditions that need to be applied
-    @notes = closed_condition(Note.all)
-    @notes = @notes.joins(:comments).where("to_tsvector('english', note_comments.body) @@ plainto_tsquery('english', ?)", params[:q])
+    if @user
+      @notes = @user.notes
+      @notes = closed_condition(@notes)
+    elsif params[:display_name] || params[:id]
+      # Return an error message because obviously the user could not be found
+      raise OSM::APIBadUserInput, "The user could not be found"
+    else
+      @notes = closed_condition(Note.all)
+    end
+
+    # Filter by a given string
+    if params[:q]
+      @notes = @notes.joins(:comments)
+      @notes = if @user
+                 @notes.where("to_tsvector('english', comments_notes.body) @@ plainto_tsquery('english', ?)", params[:q])
```

Comment:
Does the file really change name to `comment_notes` in this case? That is kind of worrying and makes this part of the code very messy... I wonder if we should always start from `Note.all` and then filter by user instead of coming at the notes from the user - that should hopefully avoid having to make this condition extra complicated.

Reply:
I definitely think there is a case for starting form `Note.all` and then using doing the user filter as `.joins(:comments).where(:note_comments => { :author => @user })` so that we can do this condition in one consistent way.

Reply:
> and then using doing the user filter as `.joins(:comments).where(:note_comments => { :author => @user })`

Well, but there is no `author` existing in the table: https://github.com/openstreetmap/openstreetmap-website/blob/master/app/models/note_comment.rb#L9-L10 we can only search by `author_ip` or `author_id`... Is the latter one maybe achievable through `.joins(:comments).where(:note_comments => { :author_id => @user.id })`?

Reply:
Sorry, that's what I meant. I was playing with with various things and copied the wrong one - you don't need `.id` on the user object though as rails will do that anyway.